internal/services/appservice/linux_web_app_resource.go

```diff
@@ -402,6 +402,22 @@ func (r LinuxWebAppResource) Create() sdk.ResourceFunc {
 
 			metadata.SetID(id)
 
+			if siteConfigAppSetting := siteConfig.AppSettings; siteConfigAppSetting != nil && len(*siteConfigAppSetting) > 0 {
+				if webApp.AppSettings == nil {
+					webApp.AppSettings = make(map[string]string)
+				}
+				for _, pair := range *siteConfigAppSetting {
+					if pair.Name == nil || pair.Value == nil {
+						continue
+					}
+					if *pair.Name == "DOCKER_REGISTRY_SERVER_URL" ||
+						*pair.Name == "DOCKER_REGISTRY_SERVER_USERNAME" ||
+						*pair.Name == "DOCKER_REGISTRY_SERVER_PASSWORD" {
+						webApp.AppSettings[*pair.Name] = *pair.Value
+					}
+				}
+			}
+
```

Comment:
I think this only address part of a larger problem here? Other evaluated settings may also be dropped from the map? I suspect what we should do here is extend the func `ExpandAppSettingsForUpdate` to take both maps and combine them to ensure everything is caught?

Reply:
Any KVP that are updated via the `siteConfig` api needs to be updated using the APPSETTING dedicated POST API(https://learn.microsoft.com/en-us/rest/api/appservice/web-apps/list-application-settings?view=rest-appservice-2022-03-01). I only updated  the docker one to follow one issue in one PR. 

You are correct about extending the ExpandAppSettingsForUpdate to merge the KVPs